### ISSUE TITLE
model_discriminator

### ISSUE THREAD

**Issue by CK11111** [jiwei0921/DCF#4]
作者您好，我不知道为什么，我训练出来的discriminator网络似乎并不能很好的划分质量好的坏的深度图。
在demo_train_pre.py中的计算IOU部分
res_r = torch.sigmoid(res_r)
res_d = torch.sigmoid(res_d)
res_r_copy = res_r.detach().squeeze().cpu().numpy().astype('int32')
 res_d_copy = res_d.detach().squeeze().cpu().numpy().astype('int32')
“.astype('int32')“直接对0~1取整，似乎会破坏res_r_copy 和res_d_copy 的图像质量，导致大量像素为0。
不知道是不是我理解有误。。。。希望作者指点一下。

**Comment by jiwei0921** [comment]
非常感谢你的关注和使用。
训练出来的discriminator确实确实存在不稳定的结果。但通过加权求和后的结果应该是有用的，对于SOD。

感谢你提的这个问题，我调试了一下 “.astype('int32')“ 这个，确实是我当时写的一个bug。我将马上修复它。
